Added the proper way to make an HMI file for ESPhome

AalianKhan opened the pull request:
I had great difficulties using this component and added how to properly make a display file so it can communicate to the microcontroller properly. Love your work 👍

## Description:


**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>
**Pull request in [esphome-core](https://github.com/esphome/esphome-core) with C++ framework changes (if applicable):** esphome/esphome-core#<esphome-core PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

AalianKhan committed:
Added the proper way to make an HMI file for ESPhome

I had great difficulties using this component and added how to properly make a display file so it can communicate to the microcontroller properly. Love your work 👍

AalianKhan commented:
I have the images but can't upload them because of permissions

OttoWinter committed:
:pencil2: Small fixes

OttoWinter reviewed (approved):
Good 👍 

As for the bkcmd, I will change it so that is set directly on boot by esphome itself - shouldn't interfere with anything else (https://github.com/esphome/esphome-core/pull/523/files#diff-854204f562df2daa82ef0efcf94d12d2R18)

AalianKhan commented:
Can you also change the baud rate to 115200 so the communication is faster

AalianKhan commented:
And no one will have to play with it to make work

AalianKhan committed:
minor spelling mistake

AalianKhan committed:
changed the the correct way to make HMI file

apparently, you have to enable both touch and release event so the microcontroller knows when the binary sensor is on/pressed and off/released. if you don't do this then any automation related to the button will only run once.

OttoWinter commented on components/display/nextion.rst:
```diff
@@ -47,14 +47,33 @@ Configuration variables:
+
+Now in ``Preinitialize Event`` You have to type ``bkcmd=3``
+
+.. figure:: images/nextion-HMI1.jpg
```
This image does not exist?

AalianKhan replied:
it says you don't have permissions to pulish images. since we are making ``bkcmd=3`` publish at start we won't need that.

AalianKhan replied:
here is the image that I want to display
![nextion_editor_2019-02-26_16-36-36](https://user-images.githubusercontent.com/39784853/53448322-21456200-39e5-11e9-8ad5-59b41afb6d01.png)

OttoWinter reviewed (commented):
No baud_rate won't be changed to something higher. The reason is that the UART for ESP8266 is a software based one, which can't handle very high baud rates like 115kbaud.

AalianKhan committed:
Update nextion.rst

OttoWinter committed:
Merge remote-tracking branch 'origin/current' into patch-1

OttoWinter committed:
Lint
